### PR TITLE
fix: (homepage) Harvest card text shows collect farm info even when only manual cake pool staked

### PR DESCRIPTION
**Pull request by memoyil**
To review:

https://deploy-preview-1887--pancakeswap-dev.netlify.app/

To reproduce:

Manual Cake pool also comes with farmsWithStaked data we should seperate it to show accurate info.

1. Go to homepage
2. Connect wallet with only has cake pool and no farms staked (or both)
3. Check text in harvest card at top
4. Text says you have farm or farms to collect but should contain information about CAKE pool instead.

### CODE REVIEW

File: src/views/Home/components/UserBanner/HarvestCard.tsx
Comment (Chef-Cheems) [inline src/views/Home/components/UserBanner/HarvestCard.tsx:52]: I just noticed that `preText` is always gonna be empty string (cause split does not return the separator in the resulting array). 
So this line can be replaced with
```javascript
const [, toCollectText] = earningsText.split(earningsBusd.toString())
```
And this block below can be removed (cause a) it never renders b) there is no point in it anyway, the Balance is showing the number)
```jsx
{preText && (
  <Text mb="4px" color="textSubtle">
    {preText}
  </Text>
)}
```

Can you do it in this PR please? 🙂

File: src/views/Home/components/UserBanner/HarvestCard.tsx
Comment (Chef-Cheems) [inline src/views/Home/components/UserBanner/HarvestCard.tsx:31]: This will show empty text, so just show `$0` with no context (while loading or when user has no stakes).
It could be changed to this:
```javascript
let earningsText = t('0 to collect') // needs to be added to translations.json
```
And then in the code below some change is needed too because `earningsBusd.toString()` might be `NaN` while loading. If user has real `0` `earningsBusd.toString()` will be `0` (just confirmed), so no need to check for that.
```javascript
const [preText, toCollectText] = earningsText.split(earningsBusd.isNaN() ? '0' : earningsBusd.toString())
```
and `preText` is not really shown so it doesn't matter, this `0` will never be displayed (see other comments about removing `preText`)

**Review by Chef-Cheems — CHANGES_REQUESTED** [review]
Thanks!
Couple of minor details, but after that its good to go :)

To be honest the whole `%earningsBusd% to collect...` seems a bit redundant since we never actually render `%earningsBusd%` part. I'm afraid it might cause troubles in non-English language, but let's keep it this way for now.

File: src/views/Home/components/UserBanner/HarvestCard.tsx
Comment (memoyil) [reply src/views/Home/components/UserBanner/HarvestCard.tsx:52]: But for example in japan translation %earningsBusd% is in the middle of the translated text, preText should not be empty string there.

`"%count%%farms%から%earningsBusd%を回収"`

Comment (Chef-Cheems) [reply src/views/Home/components/UserBanner/HarvestCard.tsx:52]: In case of those languages pretext wouldn't work anyway, cause the logic works only for English, heh.

Comment (memoyil) [reply src/views/Home/components/UserBanner/HarvestCard.tsx:52]: Why? It is working right now I think

File: src/views/Home/components/UserBanner/HarvestCard.tsx
Comment (memoyil) [reply src/views/Home/components/UserBanner/HarvestCard.tsx]: Done

File: src/views/Home/components/UserBanner/HarvestCard.tsx
Comment (memoyil) [reply src/views/Home/components/UserBanner/HarvestCard.tsx]: Then I am waiting for your approval :)

File: src/views/Home/components/UserBanner/HarvestCard.tsx
Comment (hachiojidev) [inline src/views/Home/components/UserBanner/HarvestCard.tsx:51]: It is unfortunate our translation system doesn't have an intuitive way to handle plurals. Can you move this to a helper file or somewhere out of this component. This is quite a lot of logic to just handle text.

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: src/views/Home/components/UserBanner/HarvestCard.tsx
Comment (memoyil) [reply src/views/Home/components/UserBanner/HarvestCard.tsx]: Done

**Review by hachiojidev — APPROVED** [review]

**Pull request merged**
